runtime/src/accounts_db.rs

```diff
@@ -3089,56 +3131,148 @@ impl AccountsDb {
         recycle_stores_write_elapsed.as_us()
     }
 
+    /// Purges every slot in `removed_slots` from both the cache and storage. This includes
+    /// entries in the accounts index, cache entries, and any backing storage entries.
     fn purge_slots_from_cache_and_store<'a>(
         &'a self,
-        can_exist_in_cache: bool,
         removed_slots: impl Iterator<Item = &'a Slot>,
         purge_stats: &PurgeStats,
     ) {
-        let mut remove_storages_elapsed = Measure::start("remove_storages_elapsed");
-        let mut all_removed_slot_storages = vec![];
+        let mut remove_cache_elapsed_across_slots = 0;
         let mut num_cached_slots_removed = 0;
         let mut total_removed_cached_bytes = 0;
-        let mut total_removed_storage_entries = 0;
-        let mut total_removed_stored_bytes = 0;
         for remove_slot in removed_slots {
+            // This function is only currently safe with respect to `flush_slot_cache()` because
+            // both functions run serially in AccountsBackgroundService.
+            let mut remove_cache_elapsed = Measure::start("remove_cache_elapsed");
             if let Some(slot_cache) = self.accounts_cache.remove_slot(*remove_slot) {
                 // If the slot is still in the cache, remove the backing storages for
                 // the slot and from the Accounts Index
-                if !can_exist_in_cache {
-                    panic!("The removed slot must alrady have been flushed from the cache");
-                }
                 num_cached_slots_removed += 1;
                 total_removed_cached_bytes += slot_cache.total_bytes();
                 self.purge_slot_cache(*remove_slot, slot_cache);
-            } else if let Some((_, slot_removed_storages)) = self.storage.0.remove(&remove_slot) {
+                remove_cache_elapsed.stop();
+                remove_cache_elapsed_across_slots += remove_cache_elapsed.as_us();
+            } else {
```

Comment:
maybe time to update a part of my masterpiece? lol: 

https://github.com/solana-labs/solana/blob/44831c18d29f5c43feb0ffdb0abe258be34d4a99/runtime/src/accounts_db.rs#L2563-L2581

Reply:
updated, have to keep this beauty fresh!